modin/pandas/base.py

```diff
@@ -2315,6 +2315,17 @@ def round(self, decimals=0, *args, **kwargs):
             query_compiler=self._query_compiler.round(decimals=decimals, **kwargs)
         )
 
+    def _to_datetime(self, **kwargs):
+        from .dataframe import DataFrame
+
+        if not isinstance(self, DataFrame):
```

Comment:
What are your thoughts on having this code in `dataframe.py`? If we are checking for children types in the parent class, it is often worth putting the code in the child class in my opinion.

Reply:
From my point of view this code looks pretty good despite of that we are checking for children types here. Of course, we might move the whole method `_to_datetime` to `dataframe.py` and do the same logic as here. However, I don't think it would be correctly, because both `Series` and `DataFrame` are children of `BasePandasDataset` and they should call method from base class, that is call `_to_datetime` from `BasePandasDataset`. If we want to move just the code of checking, it will need to add an additional method for `DataFrame`. This approach is not pretty good as well, as I think. So, I'd suggest keep this code. What do you think?

Reply:
I think it's better to have them separate from a code management perspective. Having a `_to_datetime` method in `DataFrame` and in `Series` would keep the implementations specific to those objects.

The `if not isinstance` check forces the parent to be aware of the types of the child and make decisions, which I believe is an antipattern.

Reply:
Oh, from the first message of this thread I thought that you asked on having the `_to_datetime` method in `dataframe.py` only and similar logic for it as in base class.
Yes, we can separate method for each object.
Great point on antipattern!